packages/main/src/Menu.ts

```diff
@@ -437,26 +444,29 @@ class Menu extends UI5Element {
 	_createSubMenu(item: MenuItem, openerId: string) {
 		const ctor = this.constructor as typeof Menu;
 		const subMenu = document.createElement(ctor.getMetadata().getTag()) as Menu;
-		const fragment = document.createDocumentFragment();
 
 		subMenu._isSubMenu = true;
 		subMenu.setAttribute("id", `submenu-${openerId}`);
 		subMenu._parentMenuItem = item;
 		subMenu.busy = item.busy;
 		subMenu.busyDelay = item.busyDelay;
-		const subItems = item.children;
-		let clonedItem,
-			idx;
-		for (idx = 0; idx < subItems.length; idx++) {
-			clonedItem = subItems[idx].cloneNode(true);
-			fragment.appendChild(clonedItem);
-		}
-		subMenu.appendChild(fragment);
+		this._cloneItems(item, subMenu);
 		this.staticAreaItem!.shadowRoot!.querySelector(".ui5-menu-submenus")!.appendChild(subMenu);
 		item._subMenu = subMenu;
 	}
 
+	_cloneItems(item: MenuItem, menu: Menu) {
+		for (let i = 0; i < item.items.length; ++i) {
+			const clonedItem = item.items[i].cloneNode(true);
+			menu.appendChild(clonedItem);
+		}
+	}
+
 	_openItemSubMenu(item: MenuItem, opener: HTMLElement, actionId: string) {
+		const mainMenu = this._findMainMenu(item);
+		mainMenu.fireEvent("before-open", {
+			item,
```

Comment:
"item": item,

Reply:
This is a short syntax automatically recommended for when the property name and value have the same identifier.